quiz/quiz.go

```diff
@@ -0,0 +1,168 @@
+/*
+	quiz defines a quiz and its questions and provides methods for starting and processing quizzes as well as reading in
+    a Quiz from file.
+*/
+package quiz
+
+import (
+	"bufio"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+)
+
+// A Quiz struct is a collection of up to 100 Questions, the name of the file containing the quiz, the number of
+// questions in the quiz and the number of questions the user got correct.
+type Quiz struct {
+	Questions        []Question
+	Filename         string
+	TotalQuestions   int
+	CorrectQuestions int
+}
+
+// NewQuiz creates a new Quiz struct based on the information in the file at the given file name
+func NewQuiz(filename string) (*Quiz, error) {
+	fmt.Println("\nGetting quiz using file:\t", filename)
+
+	// open the file and save Question to a Quiz struct
+	csvFile, err := os.Open(filename)
+
+	if err != nil {
+		fmt.Println("Error opening CSV file:\t", err)
+		return nil, err
+	}
+
+	defer func() {
+		if csvErr := csvFile.Close(); csvErr != nil {
+			fmt.Println(csvErr)
+		}
+	}()
+
+	reader := csv.NewReader(bufio.NewReader(csvFile))
+
+	var questions []Question
+	// loop until end of file or error is reached
+	for {
+		record, ioErr := reader.Read()
+		if ioErr == io.EOF {
+			break
+		}
+		if ioErr != nil {
+			fmt.Println("Error reading from file:\t", ioErr)
+			return nil, ioErr
+		}
+
+		questions = append(questions, Question{
+			QuestionText: record[0],
+			Answer:       record[1],
+		})
+	}
+
+	return &Quiz{
+		Questions:        questions,
+		Filename:         filename,
+		TotalQuestions:   len(questions),
+		CorrectQuestions: 0,
+	}, err
+}
+
+// Start begins the user-input portion of the quiz. Each Question is presented to the user and they enter answers via
+// stdin with "\n" being used to specify the end of an answer.
+func (q *Quiz) Start(maxQuizTime time.Duration, random bool) error {
+	fmt.Printf("\nYou will have %v seconds to answer %v questions.\nPress [Enter] to Start the quiz.\n",
+		maxQuizTime, q.TotalQuestions)
+
+	reader := bufio.NewReader(os.Stdin)
+	_, err := reader.ReadString('\n')
+	if err != nil {
+		return err
+	}
+
+	c := make(chan string)
```

Comment:
A better name would be `completed`.  I think we should be able to make this channel a bool rather than a string, since we are only checking if the string is equal to "success" down below.